pkg/manager/manager_test.go

```diff
@@ -1044,6 +1045,54 @@ var _ = Describe("manger.Manager", func() {
 				}))).NotTo(Succeed())
 			})
 
+			It("should not return runnables context.Canceled errors", func() {
+				Expect(options.Logger).To(BeZero(), "this test overrides Logger")
+
+				var log struct {
+					sync.Mutex
+					messages []string
+				}
+				options.Logger = funcr.NewJSON(func(object string) {
+					log.Lock()
+					log.messages = append(log.messages, object)
+					log.Unlock()
+				}, funcr.Options{})
+
+				m, err := New(cfg, options)
+				Expect(err).NotTo(HaveOccurred())
+				for _, cb := range callbacks {
+					cb(m)
+				}
+
+				// Runnables may return ctx.Err() as shown in some [context.Context] examples.
+				started := make(chan struct{})
+				Expect(m.Add(RunnableFunc(func(ctx context.Context) error {
+					close(started)
+					<-ctx.Done()
+					return ctx.Err()
+				}))).To(Succeed())
+
+				stopped := make(chan error)
+				ctx, cancel := context.WithCancel(context.Background())
+				go func() {
+					stopped <- m.Start(ctx)
+				}()
+
+				// Wait for runnables to start, signal the manager, and wait for it.
+				<-started
+				cancel()
+				Expect(<-stopped).To(Succeed())
+
+				// TODO: Diagnose why LeaderElection races after Start() returns.
+				if options.LeaderElection {
+					time.Sleep(time.Second)
+				}
+
```

Comment:
The reason this races is that you access the log output without acquiring the lock here. I am actually surprised the sleep fixed that. Proper fix would be this:
```suggestion
				log.Lock()
				defer log.Unlock()


```

Reply:
This is called after `m.Start` has returned. Is `m` expected to log in that case?

Reply:
I like `Lock` more than `Sleep`. I'll rebase and try that out.